### ISSUE TITLE
[bug] csx files containing more than one #load command are ignored, and have no intellisense

### ISSUE THREAD

**Issue by jmcginty** [OmniSharp/omnisharp-roslyn#512]
example: create a main.csx with these contents:

```
#load Teacher.csx
#load Student.csx

Console.WriteLine("hello");
```

[INFORMATION:OmniSharp.MSBuild.MSBuildProjectSystem] No solution files found in 'c:\FTP Samples\New folder'
[INFORMATION:OmniSharp.ScriptCs.ScriptCsProjectSystem] Detecting CSX files in 'c:\FTP Samples\New folder'.
[INFORMATION:OmniSharp.ScriptCs.ScriptCsProjectSystem] Found 3 CSX files.
[ERROR:OmniSharp.ScriptCs.ScriptCsProjectSystem] c:\FTP Samples\New folder\main.csx will be ignored due to the following error:
System.InvalidOperationException: **This submission already references another submission project.**
   at Microsoft.CodeAnalysis.Solution.CheckNotSecondSubmissionReference(ProjectId projectId, ProjectId toProjectId)
   at Microsoft.CodeAnalysis.Solution.AddProjectReference(ProjectId projectId, ProjectReference projectReference)
   at Microsoft.CodeAnalysis.Workspace.OnProjectReferenceAdded(ProjectId projectId, ProjectReference projectReference)
   at OmniSharp.ScriptCs.ScriptCsProjectSystem.Initalize(IConfiguration configuration)
[INFORMATION:OmniSharp.Startup] Solution has finished loading

**Comment by gaborv** [comment]
Looking into the code of ScriptCs ProjectSystem, the problem here seems a little bit bigger: the current csx support seems to be pretty basic. I am trying to extend it to consider all .csx files in all folders and properly go through the whole hierarchy of #load references.

Will post here if I can get a working build together.

**Comment by gaborv** [comment]
Alright, so here is where I could get in a day:
https://github.com/gaborv/omnisharp-roslyn/tree/vscode

This branch works with the current live VS.Code plugin. The end result is far from being perfect, but at least it does not crash on multiple loads. I could not figure out yet what makes roslyn see stuff in some (#load) referenced project but not others.

**Comment by filipw** [comment]
this is cool thanks. yes, the original implementation only looked at top level files rather than recurse over all of them.

looking at your code the nugets/script packs shouldn't be affected cause there is only one set of them anyway even if multiple levels of CSX are available, and they are equally available to each of them (represented in Roslyn by the same set of metadatareferences to every "csx-specific project" object)

**Comment by khellang** [comment]
Fixed by #578?

**Comment by filipw** [comment]
Fixed by #578
